derive/src/lib.rs

```diff
@@ -101,12 +101,23 @@ impl TypeInfoImpl {
         };
         let docs = self.generate_docs(&self.ast.attrs);
 
+        let replaces = self.attrs.replace_segments().map(|r| {
+            let search = r.search();
+            let replace = r.replace();
+
+            quote!(( #search, #replace ))
+        });
+
         Ok(quote! {
             impl #impl_generics #scale_info::TypeInfo for #ident #ty_generics #where_clause {
                 type Identity = Self;
                 fn type_info() -> #scale_info::Type {
                     #scale_info::Type::builder()
-                        .path(#scale_info::Path::new(::core::stringify!(#ident), ::core::module_path!()))
+                        .path(#scale_info::Path::new_with_replace(
+                            ::core::stringify!(#ident),
```

Comment:
Why not pass the ident as `syn::Ident`? I think it would be cleaner than calling `.split("::")` subsequently.

Reply:
We call `split` on the module path that comes out of the `module_path!` macro. Or I maybe misunderstand you?

Reply:
Oh, yea :man_facepalming: